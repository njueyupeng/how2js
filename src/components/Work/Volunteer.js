import React from 'react'
import PropTypes from 'prop-types'

const Volunteer = (props) => {
  const { theme, organization, duration, role } = props

  return (
    <React.Fragment>
      <div className="volunteer">
        <div className="information">
          <div className="organization">{organization}</div>
          <div className="duration">{duration}</div>
        </div>
        <div className="role">{role}</div>
      </div>

      {/* --- STYLES --- */}
      <style jsx global>
        {`
          .volunteer {
            padding-bottom: 15px;
          }
          .information {
            display: flex;
            flex-direction: row;
            flex-wrap: nowrap;
            justify-content: space-between;
            align-content: center;
          }
          .organization {
            font-size: 1.2rem;
            line-height: 1.42857;
            font-weight: 600;
            margin-bottom: 2px;
          }
          .duration {
            font-size: 1rem;
            color: ${theme.color.neutral.gray.g};
            font-weight: 550;
          }
          .role {
            display: flex;
            flex-direction: row;
            flex-wrap: nowrap;
            align-content: center;
          }
        `}
      </style>
    </React.Fragment>
  )
}

Volunteer.propTypes = {
  theme: PropTypes.object.isRequired,
  organization: PropTypes.string.isRequired,
  role: PropTypes.string.isRequired,
  duration: PropTypes.string.isRequired,
}

export default Volunteer
