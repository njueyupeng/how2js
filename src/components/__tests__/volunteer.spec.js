import React from 'react'
import { create } from 'react-test-renderer'
import Volunteer from '../Work/Volunteer'
const theme = require('../../theme/theme.json')

describe('Volunteer Component', () => {
  it('renders correctly', () => {
    const tree = create(
      <Volunteer
        theme={theme}
        organization="Milwaukee Area Technical College"
        role="ITDEV Advisory Board Member"
        duration="Jan 2015 - Present"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
