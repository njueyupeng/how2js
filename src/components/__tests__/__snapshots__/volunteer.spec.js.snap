// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Volunteer Component renders correctly 1`] = `
Array [
  <div
    className="volunteer"
  >
    <div
      className="information"
    >
      <div
        className="organization"
      >
        Milwaukee Area Technical College
      </div>
      <div
        className="duration"
      >
        Jan 2015 - Present
      </div>
    </div>
    <div
      className="role"
    >
      ITDEV Advisory Board Member
    </div>
  </div>,
  <style
    global={true}
    jsx={true}
  >
    
          .volunteer {
            padding-bottom: 15px;
          }
          .information {
            display: flex;
            flex-direction: row;
            flex-wrap: nowrap;
            justify-content: space-between;
            align-content: center;
          }
          .organization {
            font-size: 1.2rem;
            line-height: 1.42857;
            font-weight: 600;
            margin-bottom: 2px;
          }
          .duration {
            font-size: 1rem;
            color: #2c3e50;
            font-weight: 550;
          }
          .role {
            display: flex;
            flex-direction: row;
            flex-wrap: nowrap;
            align-content: center;
          }
        
  </style>,
]
`;
